src/server/conn.rs

```diff
@@ -0,0 +1,180 @@
+use crate::{rt::tokio_executor::TokioExecutor, util::PrependAsyncRead};
+use http::{Request, Response};
+use http_body::Body;
+use hyper::{
+    body::Incoming,
+    server::conn::{http1, http2},
+    service::Service,
+};
+use std::{error::Error as StdError, io::Cursor, marker::Unpin};
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite};
+
+const H2_PREFACE: &[u8] = b"PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n";
+
+/// Http1 or Http2 connection.
+pub struct AutoConnection {
```

Comment:
It looks like this is a builder. I'm curious about your opinion, what do you think about `server::conn::auto::Builder`?

Reply:
I was a bit lazy there. I think `Builder` api makes more sense.